[EDGE CIE-5316] c8yedge tool documentation -- DON'T MERGE OR DELETE

raghuvamsi-d opened the pull request:
- Documented the `c8yedge` tool
- Split the `install-edge` into two files one for installing with `c8yedge` tool and another for installing with the edge operator.
- Released the `c8yedge` tool as GA

raghuvamsi-d committed:
c8yedge tol documentation

raghuvamsi-d committed:
minor tweaks

BeateRixen commented on content/edge-kubernetes/quickstart.md:
```diff
@@ -6,56 +6,123 @@ sector:
 ---
 
-This section helps you to quickly install Edge on a [Lightweight Kubernetes (K3s)](https://docs.k3s.io/installation) cluster with default options. For detailed instructions, see [Installing Edge on Kubernetes](/edge-kubernetes/installing-edge-on-k8/).
+This section helps you to quickly install Cumulocity IoT Edge on a [Lightweight Kubernetes (K3s)](https://docs.k3s.io/installation) cluster using the `c8yedge` installer CLI. This tool automates the setup of K3s, Helm, Edge Operator, and the Edge instance itself.
```
```suggestion
This section helps you to quickly install {{< product-c8y-iot >}} Edge on a [Lightweight Kubernetes (K3s)](https://docs.k3s.io/installation) cluster using the `c8yedge` installer CLI. This tool automates the setup of K3s, Helm, Edge Operator, and the Edge instance itself.
```

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
```diff
@@ -6,56 +6,123 @@ sector:
+This section helps you to quickly install Cumulocity IoT Edge on a [Lightweight Kubernetes (K3s)](https://docs.k3s.io/installation) cluster using the `c8yedge` installer CLI. This tool automates the setup of K3s, Helm, Edge Operator, and the Edge instance itself.
 
-1. Verify that your hardware meets the requirements specified in [Prerequisites](/edge-kubernetes/installing-edge-on-k8/#prerequisites).
+## 1. Download and install the `c8yedge` CLI
```
```suggestion
### 1. Download and install the `c8yedge` CLI
```

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
```diff
@@ -6,56 +6,123 @@ sector:
-      USER_NAME=$(whoami)
-      USER_HOME=$(eval echo ~${USER_NAME})
-      K3S_VERSION=v1.32.3+k3s1
+## 2. View available commands
```
```suggestion
### 2. View available commands
```

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
````diff
@@ -6,56 +6,123 @@ sector:
-   '
-   ```
+```shell
+This is a CLI tool for installation and maintaining of
````
either "installing and maintaining" or "installation and maintenance"

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
````diff
@@ -6,56 +6,123 @@ sector:
-   ```
+```shell
+This is a CLI tool for installation and maintaining of
+Cumulocity Edge on Kubernetes.
````
```suggestion
{{< product-c8y-iot >}} Edge on Kubernetes.
```

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
```diff
@@ -6,56 +6,123 @@ sector:
+Available Commands:
+  completion  Generate the autocompletion script for the specified shell
+  help        Help about any command
+  install     Install Cumulocity Edge on Kubernetes
```
```suggestion
  install     Install {{< product-c8y-iot >}} Edge on Kubernetes
```

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
```diff
@@ -6,56 +6,123 @@ sector:
+  completion  Generate the autocompletion script for the specified shell
+  help        Help about any command
+  install     Install Cumulocity Edge on Kubernetes
+  remove      Uninstall Cumulocity Edge and Kubernetes
```
```suggestion
  remove      Uninstall {{< product-c8y-iot >}} Edge and Kubernetes
```

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
```diff
@@ -6,56 +6,123 @@ sector:
+  help        Help about any command
+  install     Install Cumulocity Edge on Kubernetes
+  remove      Uninstall Cumulocity Edge and Kubernetes
+  version     Default version of Cumulocity Edge on Kubernetes to install, modify it with --version flag when calling install command
```
```suggestion
  version     Default version of {{< product-c8y-iot >}} Edge on Kubernetes to install, modify it with --version flag when calling install command
```

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
````diff
@@ -6,56 +6,123 @@ sector:
+```
 
-6. Run the command below to apply Edge CR ([c8yedge-sample.yaml](/files/edge-k8s/c8yedge-sample.yaml)) for installing Edge version **{{< c8y-edge-version >}}** named **c8yedge** with the domain **myown.iot.com**.
+## 3. check the version before we start (optional)
````
```suggestion
### 3. Check the version before starting (optional)
```

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
````diff
@@ -6,56 +6,123 @@ sector:
-   ```shell
-   kubectl apply -f {{< link-c8y-doc-baseurl >}}files/edge-k8s/c8yedge-sample.yaml
-   ```
+To show the version of cli, run:
````
```suggestion
To show the version of the CLI, run:
```

BeateRixen commented on content/edge-kubernetes/quickstart.md:
````diff
@@ -6,56 +6,123 @@ sector:
+Expected output:
+
+```shell
+Cumulocity Edge installer version: 2025.0.2
````
```suggestion
{{< product-c8y-iot >}} Edge installer version: 2025.0.2
```

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
````diff
@@ -6,56 +6,123 @@ sector:
+```
+
+
+## 4. Install Edge
````
```suggestion
### 4. Install Edge
```

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
```diff
@@ -6,56 +6,123 @@ sector:
+
+Run the install command and follow the prompts. You can either pass flags explicitly or enter them interactively.
+
+### Option 1: Interactive
```
```suggestion
#### Option 1: Interactive
```

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
````diff
@@ -6,56 +6,123 @@ sector:
+Registry Password: <YourPassword>
+```
+
+### Option 2: With Flags
````
```suggestion
#### Option 2: With flags
```

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
````diff
@@ -6,56 +6,123 @@ sector:
+```
+
+
+Once complete, you will see the operator and edge deployment running:
````
```suggestion
Once complete, you will see the operator and Edge deployment running:
```

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
````diff
@@ -6,56 +6,123 @@ sector:
+...
+```
+
+## 5. What’s next?
````
```suggestion
### 5. What’s next?
```

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
````diff
@@ -6,56 +6,123 @@ sector:
+{{< /c8y-admon-important >}}
+```shell
+c8yedge remove
+Do You really want to remove Cumulocity Edge? This includes all platform data and will be non recoverable. [yes/no]: yes
````
```suggestion
Do You really want to remove {{< product-c8y-iot >}} Edge? This includes all platform data and will be non recoverable. [yes/no]: yes
```

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
```diff
@@ -6,56 +6,123 @@ sector:
+c8yedge remove
+Do You really want to remove Cumulocity Edge? This includes all platform data and will be non recoverable. [yes/no]: yes
+
+To confirm Cumulocity Edge uninstallation use this random value: <random-text> , to cancel type any other string.
```
```suggestion
To confirm {{< product-c8y-iot >}} Edge uninstallation use this random value: <random-text> , to cancel type any other string.
```

raghuvamsi-d replied:
Addressed

BeateRixen commented on content/edge-kubernetes/quickstart.md:
```diff
@@ -6,56 +6,123 @@ sector:
+Do You really want to remove Cumulocity Edge? This includes all platform data and will be non recoverable. [yes/no]: yes
+
+To confirm Cumulocity Edge uninstallation use this random value: <random-text> , to cancel type any other string.
+Confirm by typing the random value from above sentence: <random-text>
```
```suggestion
Confirm by typing the random value from the above sentence: <random-text>
```

raghuvamsi-d replied:
Good catch, it's a miss from our CLI itself, Making the changes here as well as in the product. Many thanks!

raghuvamsi-d committed:
CR:1

raghuvamsi-d committed:
quick start refactor

raghuvamsi-d committed:
fixing doc error

raghuvamsi-d committed:
edge-sample adoption for installer

raghuvamsi-d committed:
Merge branch 'release/y2025' into edge-installer-doc-pre-release

rthrippleton commented on content/edge-kubernetes/quickstart.md:
````diff
@@ -6,56 +6,54 @@ sector:
 
-3. Run the command below to install K3s.
+```shell
+curl -sfLO https://download.{{< product-c8y-iot >}}.com/Cumulocity-Edge/Installer/c8yedge && chmod +x c8yedge && ./c8yedge install --registry-host registry.c8y.io --username <YourUsername> --password <YourPassword> --version 2025 --confirm-system-requirements yes
````
```suggestion
curl -sfLO https://download.{{< product-c8y-iot >}}.com/Cumulocity-Edge/Installer/c8yedge && chmod +x c8yedge && ./c8yedge install --registry-host registry.c8y.io --username <YourUsername> --password <YourPassword>
```
`--version` should really be discouraged. By default, we want customers to just go to the latest fix. But we know there are exceptions to that; install the exact version in production that you tested. So we should still let them know that `--version` is there, but not in the first command we recommend.
`--confirm-system-requirements` exists as a command-line argument to help automation. I'd rather not encourage it, and instead have somebody go through the "Yes, I really did check my system configuration" gate that they're prompted for.

bbyreddy replied:
--username <registry username> --password <registry password>

Also add a note or INFO box below stating something like, 
"You will receive the registry credentials along with the Edge license."

raghuvamsi-d replied:
Totally agreed on this @rthrippleton , I will change this accordingly

Done @bbyreddy , Added the changes already as suggested.

raghuvamsi-d replied:
Completed

rthrippleton commented on content/edge-kubernetes/quickstart.md:
````diff
@@ -6,56 +6,54 @@ sector:
 
-3. Run the command below to install K3s.
+```shell
+curl -sfLO https://download.{{< product-c8y-iot >}}.com/Cumulocity-Edge/Installer/c8yedge && chmod +x c8yedge && ./c8yedge install --registry-host registry.c8y.io --username <YourUsername> --password <YourPassword> --version 2025 --confirm-system-requirements yes
````
```suggestion
curl -sfLO https://download.cumulocity.com/Cumulocity-Edge/Installer/c8yedge && chmod +x c8yedge && ./c8yedge install --registry-host registry.c8y.io --username <YourUsername> --password <YourPassword> --version 2025 --confirm-system-requirements yes
```
{{< product-c8y-iot >}} is purely for branding, and may (and has in the past) change to include spaces and suchlike. So in this case, just hardcoding is safer. Looks like https://download.cumulocity.com/ appears verbatim in other parts of the Cumulocity doc, so we're in good company.

raghuvamsi-d replied:
Sure, I will change it right away.

raghuvamsi-d replied:
Completed

rthrippleton commented on content/edge-kubernetes/quickstart.md:
````diff
@@ -6,56 +6,54 @@ sector:
 
-3. Run the command below to install K3s.
+```shell
+curl -sfLO https://download.{{< product-c8y-iot >}}.com/Cumulocity-Edge/Installer/c8yedge && chmod +x c8yedge && ./c8yedge install --registry-host registry.c8y.io --username <YourUsername> --password <YourPassword> --version 2025 --confirm-system-requirements yes
````
The layout of our downloadable artifacts ought to be including the major release i.e. https://download.cumulocity.com/Cumulocity-Edge/Installer/2025/c8yedge

Because at some point in the future both 2025 and 2026 will be both supported.

raghuvamsi-d replied:
Right, This is still a tentative location. I will update it once we conclude the approach.

bbyreddy commented on content/edge-kubernetes/quickstart.md:
```diff
@@ -6,56 +6,54 @@ sector:
 
-4. Run the command below to install Helm v3.
+### 2. What’s next?
+{{< c8y-admon-info >}}
+To know your c8yedge cli version, execute **`c8yedge version`**
+{{< /c8y-admon-info >}}
```
- [x] May be this is not required. And definitely not in the what next section.
We can add such details later when we start documenting the c8yedge tool as a standlone section in the doc.

raghuvamsi-d replied:
Added in the info section, if it finds itself in odd place, I am happy remove it

bbyreddy commented on content/edge-kubernetes/quickstart.md:
```diff
@@ -6,56 +6,54 @@ sector:
-      touch /etc/sysctl.d/90-kubelet.conf  && \
-      sed -i "/^vm\.panic_on_oom=/d; /^vm\.overcommit_memory=/d; /^kernel\.panic=/d; /^kernel\.panic_on_oops=/d" /etc/sysctl.d/90-kubelet.conf && \
-      printf "vm.panic_on_oom=0\nvm.overcommit_memory=1\nkernel.panic=10\nkernel.panic_on_oops=1\n" | tee -a /etc/sysctl.d/90-kubelet.conf && \
+Once complete, you will see the operator and Edge deployment running:
```
Instead of mentioning what to expect in the kubernetes (which is already explained in the "verifying-the-edge-installation" section). I suggest we show the output of the c8yedge tool, which I guess at the end shows the event "Cumulocity Edge installation is complete in 4m58s, and it's now running version 2025.0.2-xxxx"

raghuvamsi-d replied:
Done

nickponomar-c8y commented:
@raghuvamsi-d how can we make the reader to understand that c8yedge is not the only option to install? While it is ok to have a quick guide but it should be clear that c8yedge is one of the two options to install Cumulocity Edge.
How do we communicate when to choose what?

raghuvamsi-d committed:
Change requests, refining quick start and related docs

raghuvamsi-d committed:
Merge branch 'edge-installer-doc-pre-release' of https://github.com/Cumulocity-IoT/c8y-docs into edge-installer-doc-pre-release

raghuvamsi-d committed:
Change request addressed for quick start

raghuvamsi-d committed:
Merge branch 'release/y2025' into edge-installer-doc-pre-release

raghuvamsi-d committed:
correcting short code

raghuvamsi-d committed:
Merge branch 'edge-installer-doc-pre-release' of https://github.com/Cumulocity-IoT/c8y-docs into edge-installer-doc-pre-release

raghuvamsi-d committed:
Reverting as its generic flow

raghuvamsi-d commented:
> @raghuvamsi-d how can we make the reader understand that c8yedge is not the only option to install? While it is ok to have a quick guide but it should be clear that c8yedge is one of the two options to install Cumulocity Edge. How do we communicate when to choose what?

May be we can add a note at the top of the doc stating, This is not for power users for those well versed with k8s they can use the regular installation flow (url)?

raghuvamsi-d committed:
not adding short codes for commands and urls

raghuvamsi-d committed:
Addressing review comments on installation part

raghuvamsi-d committed:
missing shell in the doc

bbyreddy committed:
[EDGE CIE-5316] edge-installer-cleanup-appliance-vm-docs (#3367)

Merged the second level "Edge Appliance VM" and "Edge on K8s" cards into a single level top-level card named "Edge".
Removed the Edge Appliance VM content (deleted the folders content\edge and static\files\edge)
Moved Edge introduction and some content which was only present in Edge Appliance VM into edge-kubernetes.
Rewired the links to the Edge Appliance VM content (from across the docs) to the content under content\edge-kubernetes
Renamed the "edge-operator" build_artifact label to "edge" and also removed the edge-appliance build artifact.
Documented the c8yedge tool
Split the install-edge into two files one for installing with c8yedge tool and another for installing with the edge operator.

bbyreddy committed:
[EDGE CIE-5316] edge-installer-cleanup-appliance-vm-docs (#3367)

Merged the second level "Edge Appliance VM" and "Edge on K8s" cards into a single level top-level card named "Edge".
Removed the Edge Appliance VM content (deleted the folders content\edge and static\files\edge)
Moved Edge introduction and some content which was only present in Edge Appliance VM into edge-kubernetes.
Rewired the links to the Edge Appliance VM content (from across the docs) to the content under content\edge-kubernetes
Renamed the "edge-operator" build_artifact label to "edge" and also removed the edge-appliance build artifact.
Documented the c8yedge tool
Split the install-edge into two files one for installing with c8yedge tool and another for installing with the edge operator.

bbyreddy committed:
Merge branch 'release/y2025' into edge-installer-doc-pre-release

bbyreddy committed:
Merge branch 'release/y2025' into edge-installer-doc-pre-release

bbyreddy committed:
Minor changes to the tool download commands and description of the help commands.

bbyreddy committed:
Minor changes to the tool download commands and description of the help commands.

raghuvamsi-d committed:
version bump

bbyreddy committed:
Some more updates

bbyreddy committed:
Update _index.md

bbyreddy committed:
Revert "Update _index.md"

This reverts commit 3c9124e8d7824181ebad6f9e02c06a07261d7058.

bbyreddy committed:
Update _index.md

bbyreddy committed:
Update _index.md

bbyreddy committed:
Update _index.md

bbyreddy committed:
Update _index.md

bbyreddy committed:
Merge branch 'edge/CIE-5316/cleanup-edge-appliance-vm-docs' into edge-installer-doc-pre-release

bbyreddy committed:
Update modify-edge.md

bbyreddy committed:
Update install-edge-with-edge-operator.md

bbyreddy committed:
Updated the chmod command

rthrippleton commented on content/edge-kubernetes/installing-edge-on-k8-bundle/install-edge-with-c8yedge-cli.md:
```diff
@@ -0,0 +1,34 @@
+---
+weight: 40
```
I think this is positioned incorrectly:
![image](https://github.com/user-attachments/assets/f21ddf4d-8762-4784-b828-dcc05ceb07c0)
You have to go through 'Configuring proxy' and 'Configuring storage' before you come to 'Installing with c8yedge tool' - those two previous sections are only really relevant to the 'power user' section.

In fact, if it's possible to do so many levels, I'd suggest changing the _nesting_:
* Installation
* * Prerequisites
* * Installing with c8yedge tool
* * Installing with Edge operator
* * * Configuring proxy
* * * Configuring storage

rthrippleton replied:
So apparently we can't nest sections that deep. But I've at least re-ordered to be better (IMO). 96540ed38

rthrippleton replied:
4b89bacf93a4393bc0b138afcc2986e8c105463f

rthrippleton commented on content/edge-kubernetes/installing-edge-on-k8-bundle/install-edge-with-c8yedge-cli.md:
````diff
@@ -0,0 +1,34 @@
+The `c8yedge` command-line tool automates the entire process of preparing the environment and installing Edge. You can download the tool from the [{{< company-c8y >}} Download Center](https://download.cumulocity.com/Cumulocity-Edge/Installer) or by running the following commands:
+
+```shell
+curl -sfL https://download.cumulocity.com/Cumulocity-Edge/Installer/c8yedge -o c8yedge
````
I don't recall, but didn't we discuss versioning this just to be futureproof? That is

https://download.cumulocity.com/Cumulocity-Edge/Installer/2025/c8yedge

bbyreddy replied:
You are right, I changed it. https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/commits/840045fcd816acc4c645c3150b4ee928ece3758c

rthrippleton commented on content/edge-kubernetes/installing-edge-on-k8-bundle/install-edge-with-c8yedge-cli.md:
```diff
@@ -0,0 +1,34 @@
+For more information about the tool, run `c8yedge --help` or `c8yedge [command] --help`.
+
+### Install Edge
+Before proceeding, ensure all [prerequisites](/edge-kubernetes/installing-edge-on-k8/#prerequisites) are met.
```
I think we also need to call out https://docs.k3s.io/installation/requirements#operating-systems here. I know that further down in the prerequisites we talk about this in "SPECIAL INSTRUCTIONS FOR K3S", but that's confusing in this case as it's aimed at users installing k3s by hand.

(In fact, I wonder if we ought to remove any mention of k3s as being special in that section - either you use c8yedge, or you pick one of many kubernetes distributions. I'll think on that.)

bbyreddy replied:
Removed the reference to single-node cluster: https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/commits/83e879f284cdabec85ebb998a1e801a37b111b22

bbyreddy replied:
> I think we also need to call out https://docs.k3s.io/installation/requirements#operating-systems here. I know that further down in the prerequisites we talk about this in "SPECIAL INSTRUCTIONS FOR K3S", but that's confusing in this case as it's aimed at users installing k3s by hand.
> 
> (In fact, I wonder if we ought to remove any mention of k3s as being special in that section - either you use c8yedge, or you pick one of many kubernetes distributions. I'll think on that.)

Addressed in https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/commits/ea6c238393b9d4391a4b74a8a9f2b0afe63b9be4

rthrippleton commented on content/edge-kubernetes/installing-edge-on-k8.md:
```diff
@@ -6,4 +6,33 @@ sector:
+      - **Name**: `c8yedge`
+      - **Company**: `Edge Bootstrap`
+      - **Domain**: `edgebootstrap.example`
+      - **Email**: `company@edgebootstrap.example`
```
So as not to alarm users away from using `c8yedge`, we should reassure them that these can all be changed later.

bbyreddy replied:
Addressed by https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/commits/210b2f6fd26554888d6f68f430c46b0eb6023b60

rthrippleton commented on content/edge-kubernetes/installing-edge-on-k8.md:
```diff
@@ -6,4 +6,33 @@ sector:
+
+  - **[Installing with the Edge Operator](#install-edge-with-edge-operator) (For users with an existing Kubernetes setup)**
+    <br/>
+    This method is suitable for users who already have a Kubernetes cluster (typically with a single worker node) and want to install Edge using the Kubernetes-native approach.
```
```suggestion
    This method is suitable for users who already have a Kubernetes cluster and want to install Edge using the Kubernetes-native approach. Even a single-node Kubernetes cluster will suffice, but is not required.
```
as a power user, I don't want to be told how I have to set up _my_ kubernetes cluster :-)

bbyreddy replied:
Resolved: https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/commits/4cef6481dd01c65061cfa90b9453a024de4a857b

rthrippleton commented on content/edge-kubernetes/installing-edge-on-k8.md:
```diff
@@ -6,4 +6,33 @@ sector:
+    This method is suitable for users who already have a Kubernetes cluster (typically with a single worker node) and want to install Edge using the Kubernetes-native approach.
+    <br/>
+    In this case, you are expected to:
+    - Set up and manage the Kubernetes cluster yourself (including K3s or any compatible Kubernetes distribution).
```
```suggestion
    - Set up and manage the Kubernetes cluster yourself (using K3s or any compatible Kubernetes distribution).
```

bbyreddy replied:
Addressed by https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/commits/16583477b55aa3a6c72651316e9a2fc282583fc1

rthrippleton commented on content/edge-kubernetes/installing-edge-on-k8.md:
```diff
@@ -6,4 +6,33 @@ sector:
+    <br/>
+    In this case, you are expected to:
+    - Set up and manage the Kubernetes cluster yourself (including K3s or any compatible Kubernetes distribution).
+    - Install any [prerequisites](/edge-kubernetes/installing-edge-on-k8/#prerequisites)
```
```suggestion
    - Do any system configuration or software installation required by the [prerequisites](/edge-kubernetes/installing-edge-on-k8/#prerequisites)
```
(a lot of those requirements are non-software)

bbyreddy replied:
Addressed by https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/commits/485b151d04a47a58ffc71043a0a1fe908e2789bf

rthrippleton commented on content/edge-kubernetes/quickstart.md:
```diff
@@ -6,56 +6,27 @@ sector:
+
+Upon successful installation, the tool will exit automatically.
+
+To sign in to Edge, refer to the [**Accessing Edge**](/edge-kubernetes/installing-edge-on-k8/#accessing-edge) section. The initial installation uses a bootstrap domain (`edgebootstrap.example`) along with a default license and email (`company@edgebootstrap.example`). Use this domain, email and the default credentials username “admin” and password “admin-pass” to access Edge for the first time. Yo ...
```
```suggestion
To sign in to Edge, refer to the [**Accessing Edge**](/edge-kubernetes/installing-edge-on-k8/#accessing-edge) section. The initial installation uses a bootstrap domain (`edgebootstrap.example`) along with a default license and email (`company@edgebootstrap.example`). Use this domain, email and the default credentials username “admin” and password “admin-pass” to access Edge for the first time. You can later update the domain and license to match your environment by following the steps outlined in [**Modifying Edge**](/edge-kubernetes/manage-edge/#modify-edge).
```
"bootstrap domain" isn't a C8Y term, it's only used informally in the installer.

rthrippleton replied:
Oh, and a similar modification to https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/files#r2114294573

bbyreddy replied:
Changed the text to completely remove the wording bootstrap: https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/commits/b1d9c46466d3ce16d92e4a8b682ef890032e3b72

rthrippleton commented on content/edge-kubernetes/installing-edge-on-k8-bundle/install-edge-with-edge-operator.md:
```diff
@@ -1,21 +1,59 @@
 
+Before you start the installation, ensure that you have fulfilled the [prerequisites](/edge-kubernetes/installing-edge-on-k8/#prerequisites) and configured the storage as described in [Configuring storage](/edge-kubernetes/installing-edge-on-k8/#configuring-storage).
+
+To begin, create a new single-node Kubernetes cluster with the Kubernetes version and the platform of your choice, and configure `kubectl` to use that cluster. See [Prerequisites](/edge-kubernetes/installing-edge-on-k8/#prerequisites) for the supported Kubernetes versions and platforms.
```
I think we're giving mixed messages here ("single-node Kubernetes cluster") - in some places we're quite soft about this requirements, in others we aren't. And a power user probably won't like this constraint.

bbyreddy replied:
Actually the second statement is not required. Fulfilling the prerequisites means the required software (including Kubernetes) and configurations are in place. I will remove it. https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/commits/cc06a91bc85503c2ab3cc55305ad710d7340b0fe

bbyreddy committed:
Updated the c8yedge download URL to keep the version for futureproofing it.

bbyreddy committed:
Updated the links to installation root.

rthrippleton commented on content/edge-kubernetes/installing-edge-on-k8-bundle/install-edge-with-edge-operator.md:
```diff
@@ -1,21 +1,59 @@
 
 Refer to [Harbor Installation and Configuration](https://goharbor.io/docs/2.11.0/install-config/) for installing Harbor and [Nexus Installation and Upgrades](https://help.sonatype.com/en/install-nexus-repository.html) for installing Nexus.
 
 After installing and configuring a private registry, ensure that all the machines (the workstation and the Kubernetes cluster nodes) which need access to the private registry can resolve its domain or host and trust the private regsitry's certificate (if it is configured with a self-signed certificate).
```
```suggestion
After installing and configuring a private registry, ensure that all the machines (the workstation and the Kubernetes cluster nodes) which need access to the private registry can resolve its domain or host and trust the private registry's certificate (if it is configured with a self-signed certificate).
```

rthrippleton commented on content/edge-kubernetes/installing-edge-on-k8-bundle/install-edge-with-edge-operator.md:
```diff
@@ -1,21 +1,59 @@
+To install the Edge operator, download and run the script, refer to a sample command below. Enter the version (`-v` option, for example, {{< c8y-edge-version >}}) you want to install, registry hostname (`-r` option) and the registry credentials you received along with the license when prompted. *Use `-h` option to display the usage details.*
+
+{{< c8y-admon-info >}}
+If you are installing Edge from a local/private registry, provide the hostname (`-r` option) as <registry-hostname>:<registry-port> and the respective credentials when prompted.
```
Needs a cross-reference to the later section about private registries - if I'm reading this and see such an enticing idea, I want to know more!

bbyreddy replied:
Addressed in https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/commits/ae42cb99816ef618af8a0e719ef51919c53c3ad4

bbyreddy committed:
updated the text to assure the users that the default bootstrap configurations can be modified later.

bbyreddy committed:
Changed the message to say even a single node cluster suffices instead of telling the power user how to setup his cluster.

Co-authored-by: Richard Thrippleton <richard.thrippleton@softwareag.com>

bbyreddy committed:
Applied the suggestion

Co-authored-by: Richard Thrippleton <richard.thrippleton@softwareag.com>

rthrippleton committed:
Typo

bbyreddy committed:
Replace "Install any [prerequisites]" with "Do any system configuration or software installation"

Co-authored-by: Richard Thrippleton <richard.thrippleton@softwareag.com>

rthrippleton committed:
Reorder sections to avoid the implication that you can do this stuff before running c8yedge

rthrippleton commented on content/edge-kubernetes/installing-edge-on-k8-bundle/install-edge-with-c8yedge-cli.md:
```diff
@@ -0,0 +1,34 @@
+
+Upon successful installation, the tool will exit automatically.
+
+To sign in to Edge, refer to the [**Accessing Edge**](/edge-kubernetes/installing-edge-on-k8/#accessing-edge) section. The initial installation uses a bootstrap domain (`edgebootstrap.example`) along with a default license and email (`company@edgebootstrap.example`). Use this domain, email and the default credentials username “admin” and password “admin-pass” to access Edge for the first time. Yo ...
```
I don't want anybody thinking that "http://edgebootstrap.example" will work for them - I want them to have to go through to Accessing Edge to find out the real way of accessing it.
```suggestion
To sign in to Edge, refer to the [**Accessing Edge**](/edge-kubernetes/installing-edge-on-k8/#accessing-edge) section. The initial installation uses a dummy domain along with a default license and email (`company@edgebootstrap.example`). Use this domain, email and the default credentials username “admin” and password “admin-pass” to access Edge for the first time. You can later update the bootstrap domain and license to match your environment by following the steps outlined in [**Modifying Edge**](/edge-kubernetes/manage-edge/#modify-edge).
```

bbyreddy replied:
Changed the text to completely remove the wording bootstrap: https://github.com/Cumulocity-IoT/c8y-docs/commit/b1d9c46466d3ce16d92e4a8b682ef890032e3b72
Please check the text and let me know if you are fine with it.

bbyreddy replied:
The Accessing Edge section uses myown.iot.com as the domain for explaining the management and edge tenant domains and how to map the IP etc..

rthrippleton replied:
I'd rather not mention the name of the domain at all - there's absolutely nothing useful a user could do with that, except mistakenly assume it should work.

You're right about Accessing Edge - I'm wondering if I should add wording for even simpler ways to access it.
* If it's on your desktop - http://localhost
* If you know the IP of the machine - http://<ip> - unless firewalling is set up.
along with the caveats that you need to name the tenant. Because it's not much of a Quickstart if you have to configure DNS or mangle hosts.

What do you think?

bbyreddy replied:
> I'd rather not mention the name of the domain at all - there's absolutely nothing useful a user could do with that, except mistakenly assume it should work.
> 
> You're right about Accessing Edge - I'm wondering if I should add wording for even simpler ways to access it.
> 
> * If it's on your desktop - http://localhost
> * If you know the IP of the machine - http:// - unless firewalling is set up.
>   along with the caveats that you need to name the tenant. Because it's not much of a Quickstart if you have to configure DNS or mangle hosts.
> 
> What do you think?

This is better than just referring to "myown.iot.com" as a domain in the Accessing Edge section. This was needed earlier as we couldn't access Edge without going through a domain (due to OAI secure flag). Since we removed it, we are better of saying use IP or domain (if domain you used can be resolved).

rthrippleton replied:
8d39fa1bd

bbyreddy committed:
Updated the text to remove the reference to bootstrap.

bbyreddy committed:
Merge branch 'edge-installer-doc-pre-release' of https://github.com/Cumulocity-IoT/c8y-docs into edge-installer-doc-pre-release

rthrippleton committed:
Nest these sections to look even better

bbyreddy committed:
Cross-linked the "local/private registry" with the offline installation section.

bbyreddy committed:
Merge branch 'edge-installer-doc-pre-release' of https://github.com/Cumulocity-IoT/c8y-docs into edge-installer-doc-pre-release

bbyreddy committed:
Removed the duplicate mention of installing single-node K8s cluster.

bbyreddy committed:
Removed the restriction of single-node cluster support from the docs.

bbyreddy committed:
More updates to the prerequisites.

bbyreddy committed:
Updated the prerequisites text.

bbyreddy committed:
"operator" in the "Edge operator" is expected to start with a lowercase "o"

bbyreddy committed:
Removed unnecessary references to myown.iot.com where not required.

bbyreddy committed:
In case of quickstart the domain used is edgebootstrap.example

bbyreddy committed:
Removed the special instructions to install K3s and any explicit references to K3s in prerequisites (except that it is in the smaple kubernetes distro list).

bbyreddy committed:
Removed reference to K3s in the upgrades.

bbyreddy committed:
Removed the references to K8s/K3s upgrade from prerequisites.

bbyreddy committed:
Updates to prerequisites

bbyreddy committed:
Merge branch 'edge/CIE-5316/c8yedge-tool-alpha-release-1' into edge/CIE-5316/cleanup-edge-appliance-vm-docs

bbyreddy committed:
Merge branch 'edge/CIE-5316/cleanup-edge-appliance-vm-docs' into edge-installer-doc-pre-release

bbyreddy committed:
Update c8yedge-sample.yaml

bbyreddy committed:
Updated the registry sync python wheel for 2025.0.2 version.

bbyreddy committed:
Merge branch 'edge/CIE-5316/c8yedge-tool-alpha-release-1' into edge/CIE-5316/cleanup-edge-appliance-vm-docs

bbyreddy committed:
Merge branch 'edge/CIE-5316/cleanup-edge-appliance-vm-docs' into edge-installer-doc-pre-release

bbyreddy committed:
Updated the docs to change the location of the c8yedge_registry_sync to Cumulocity downloads site.

bbyreddy committed:
Merge branch 'edge/CIE-5316/c8yedge-tool-alpha-release-1' into edge/CIE-5316/cleanup-edge-appliance-vm-docs

bbyreddy committed:
Merge branch 'edge/CIE-5316/cleanup-edge-appliance-vm-docs' into edge-installer-doc-pre-release

bbyreddy committed:
Update install-edge-with-edge-operator.md

rthrippleton committed:
edgebootstrap.example is very unlikely to ever be used for real - misleading

rthrippleton commented on content/edge-kubernetes/edge-operations-bundle/monitoring.md:
```diff
@@ -5,7 +5,7 @@ layout: redirect
 Edge allows for monitoring of the Edge deployment using Prometheus, an open-source project that is used for monitoring application state. See https://prometheus.io/ for detailed information on Prometheus and how to use it.
 
-The Edge operator exposes a Prometheus-compatible metrics endpoint, `https://\<domain>:3443/metrics`, where the domain is the one you specified in the Edge CR (or myown.iot.com if you followed the Quickstart installation steps).
+The Edge operator exposes a Prometheus-compatible metrics endpoint, `https://\<domain>:3443/metrics`, where the domain is the one you specified in the Edge CR (or edgebootstrap.example if you followed the Quickstart installation steps).
```
03fd41ff1

rthrippleton committed:
Emphasise the really simple way to initially access Edge

rthrippleton committed:
Centralize the credentials information, and mention both tenants

rthrippleton committed:
This was somehow fouling up the rendering of the link on the next line

rthrippleton committed:
'Edge operator' -> 'Kubernetes-native approach'

The latter is more meaningful to somebody coming to this product for the first time

rthrippleton committed:
Missed a bit from dd969fbe8e0b1e2f40a49271cff9f9534ef1fdcb

rthrippleton committed:
'with c8yedge tool' -> 'with the c8yedge tool', better English

bbyreddy committed:
Merge branch 'release/y2025' into edge/CIE-5316/cleanup-edge-appliance-vm-docs

bbyreddy committed:
Merge branch 'edge/CIE-5316/cleanup-edge-appliance-vm-docs' into edge-installer-doc-pre-release

rthrippleton commented on content/edge-kubernetes/installing-edge-on-k8-bundle/install-edge-with-c8yedge-cli.md:
```diff
@@ -0,0 +1,34 @@
+---
+weight: 20
+title: Installing with the c8yedge tool
```
cf62f1151e89e26fc66fc42d74b59e56061eee72

rthrippleton reviewed (approved):
Once you're ok with all of the changes I've injected into this PR

bbyreddy committed:
Added a Private Preview Release notes entry for c8yedge tool

rthrippleton committed:
Remove 'Edge on Kubernetes' mentions, even though I suspect these two are going to be wiped out by another PR

rthrippleton committed:
Merge branch 'edge/CIE-5316/cleanup-edge-appliance-vm-docs' into edge-installer-doc-pre-release

bbyreddy committed:
Addressed review comment.

Co-authored-by: Richard Thrippleton <richard.thrippleton@softwareag.com>

bbyreddy committed:
Updated the "On-premises deployment with Cumulocity Edge" section to be specific about the locally managed K8s.

bbyreddy committed:
Merge branch 'edge/CIE-5316/cleanup-edge-appliance-vm-docs' of https://github.com/Cumulocity-IoT/c8y-docs into edge/CIE-5316/cleanup-edge-appliance-vm-docs

bbyreddy committed:
Merge branch 'edge/CIE-5316/cleanup-edge-appliance-vm-docs' into edge-installer-doc-pre-release

bbyreddy committed:
Merge branch 'release/y2025' into edge/CIE-5316/cleanup-edge-appliance-vm-docs

bbyreddy committed:
Merge branch 'edge/CIE-5316/cleanup-edge-appliance-vm-docs' into edge-installer-doc-pre-release

rthrippleton committed:
It's 'Edge' not 'The Edge'

rthrippleton committed:
Update content/edge-kubernetes/installing-edge-on-k8-bundle/install-operator-offline.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

rthrippleton committed:
Update content/edge-kubernetes/installing-edge-on-k8-bundle/prerequisites.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

rthrippleton committed:
Update content/edge-kubernetes/k8-edge-introduction-bundle/c8y-iot-edge-k8-operator.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

rthrippleton committed:
Update content/edge-kubernetes/k8-edge-introduction-bundle/c8y-iot-edge-k8-operator.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

rthrippleton committed:
Update content/edge-kubernetes/k8-edge-introduction-bundle/c8y-iot-edge-k8-operator.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

rthrippleton committed:
Update content/edge-kubernetes/k8-edge-introduction-bundle/c8y-iot-edge-k8-operator.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

rthrippleton committed:
Update content/edge-kubernetes/k8-edge-introduction-bundle/c8y-iot-edge-k8-operator.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

rthrippleton committed:
Logistics team is not a term we use any more

bbyreddy committed:
Fixed some broken links and removed on-kubernetes suffixes fro DataHub content.

bbyreddy committed:
Merge branch 'edge/CIE-5316/cleanup-edge-appliance-vm-docs' into edge-installer-doc-pre-release

bbyreddy committed:
Update content/edge-kubernetes/k8-using-edge-bundle/developing-an-application.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

bbyreddy committed:
Merge branch 'edge/CIE-5316/cleanup-edge-appliance-vm-docs' into edge-installer-doc-pre-release

bbyreddy committed:
Merge branch 'release/y2025' into edge/CIE-5316/cleanup-edge-appliance-vm-docs

bbyreddy committed:
Merge branch 'edge/CIE-5316/cleanup-edge-appliance-vm-docs' into edge-installer-doc-pre-release

bbyreddy committed:
Update edge-2025.0.2-c8yedge-command-line-tool.md

bbyreddy committed:
Merge branch 'release/y2025' into edge-installer-doc-pre-release

rthrippleton committed:
Making the 'c8yedge' install path less confusing (#3425)

We wanted to hand-hold a bit more for users who are packaging a VM using 'c8yedge', and not confuse them with unnecessary kubernetes concepts.

bbyreddy committed:
Merge branch 'release/y2025' into edge-installer-doc-pre-release

bbyreddy committed:
Render the base url in lowercase, so that the curl downloads don't complain in surge sites.

rthrippleton committed:
Remove c8yedge-operator-install.sh now that c8yedge exists

The rationale is that a non-kubernetes-native would now use c8yedge, so we no longer need to inflict something on a kubernernetes native that is little more than a thin helm wrapper.

bbyreddy commented on content/edge-kubernetes/installing-edge-on-k8-bundle/install-edge-with-edge-operator.md:
```diff
@@ -15,37 +15,33 @@ Edge has been tested and officially supported on Kubernetes version 1.32.x, the
-If you are installing Edge from a [local/private registry](/edge-kubernetes/installing-edge-on-k8/#installing-edge-operator-offline), provide the hostname (`-r` option) as <registry-hostname>:<registry-port> and the respective credentials when prompted.
-{{< /c8y-admon-info >}}
-
+The Edge operator is available as a Helm chart in the Edge registry, and can be applied in the usual manner. First, log in to the registry
```
Do we need to separate the login and install commands? As this adds more complexity while explaining the option of "installing Edge from a [local/private registry]" as they need to change the registry name at multiple places. --username and --password can be passed to the helm install command.

rthrippleton replied:
a5754efae

bbyreddy commented on content/edge-kubernetes/installing-edge-on-k8-bundle/install-edge-with-edge-operator.md:
```diff
@@ -15,37 +15,33 @@ Edge has been tested and officially supported on Kubernetes version 1.32.x, the
+    --version={{< c8y-edge-version >}}
+    --namespace c8yedge \
+    --create-namespace \
+    --set imageCredentials.username="<your registry username>" \
+    --set imageCredentials.password="<your registry password>" \
```
We have been referring to the registry credentials as "<Edge registry username/passowrd>". You may want to change just to be consistent.

rthrippleton replied:
a5754efae

bbyreddy commented on content/edge-kubernetes/installing-edge-on-k8-bundle/install-edge-with-edge-operator.md:
````diff
@@ -15,37 +15,33 @@ Edge has been tested and officially supported on Kubernetes version 1.32.x, the
+```shell
+helm upgrade --install c8yedge-operator oci://registry.c8y.io/edge/helm-charts/cumulocity-iot-edge-operator \
+    --version={{< c8y-edge-version >}}
+    --namespace c8yedge \
````
We can exclude the `--namespace` (standard helm option) as the Edge Operator helm chart defaults it to `c8yedge` and also if someone really want to install it in a different namespace, they would know the standard helm options.

bbyreddy replied:
To avoid giving too many options to the end user, which forces us to keep repeating the below statement:

{{< c8y-admon-info >}}
Substitute the namespace name *c8yedge* in the command above with the namespace name where you have installed the Edge operator.
{{< /c8y-admon-info >}}

rthrippleton replied:
> We can exclude the `--namespace` (standard helm option) as the Edge Operator helm chart defaults it to `c8yedge` and also if someone really want to install it in a different namespace, they would know the standard helm options.

`Error: execution error at (cumulocity-iot-edge-operator/templates/validate-namespace.yaml:4:4): Invalid namespace. Specify a custom namespace using the --namespace option when installing the Helm Chart.`

rthrippleton replied:
Also, I don't think we need to tell anybody how to change the namespace - they'll see `--namespace` and know what to do.

rthrippleton committed:
Use one single command instead of two, and make the offline instructions even clearer

rthrippleton committed:
Merge branch 'release/y2025' into edge-installer-doc-pre-release

rthrippleton committed:
Unbreak the hugo build

rthrippleton committed:
Improving the installation doc based on local experience (#3493)

Making it clear that messing around with domain names and IP addresses is unnecessary in many/some cases.

rthrippleton committed:
Try to make the user flow for post-installation more readable (#3530)

* Making the post-installation configuration more straightforward
* Moving obscure reference documentation and details out of the way

bbyreddy committed:
Merge branch 'release/y2025' into edge-installer-doc-pre-release

rthrippleton committed:
Documentation for changing registry credentials (#3557)

rthrippleton committed:
LoadBalancer prerequisite (#3612)

bbyreddy committed:
[EDGE CIE-5358] Update version references in install/update commands   (#3552)

* Removed the use of short codes c8y-edge-version, c8y-edge-version-minor and c8y-edge-version-patch and renamed c8y-edge-version-major to c8y-edge-current-version.

* Corrected invalid command.

* Removed duplicate comment.

* minor text change.

* Update content/edge-kubernetes/installing-edge-on-k8-bundle/install-edge-with-edge-operator.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

* Elaborated the version selection in the upgrade section.

---------

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

rthrippleton committed:
It's now in GA
